src/derived/interpolate.js

```diff
@@ -0,0 +1,121 @@
+import {
+  cond,
+  lessThan,
+  multiply,
+  sub,
+  add,
+  divide,
+  greaterThan,
+} from '../base';
+import invariant from 'fbjs/lib/invariant';
+import AnimatedNode from '../core/AnimatedNode';
+
+function interpolateInternalSingle(value, inputRange, outputRange, offset) {
+  const inS = inputRange[offset];
+  const inE = inputRange[offset + 1];
+  const outS = outputRange[offset];
+  const outE = outputRange[offset + 1];
+  const progress = divide(sub(value, inS), sub(inE, inS));
+  return add(outS, multiply(progress, sub(outE, outS)));
+}
+
+function interpolateInternal(value, inputRange, outputRange, offset = 0) {
+  if (inputRange.length - offset === 2) {
+    return interpolateInternalSingle(value, inputRange, outputRange, offset);
+  }
+  return cond(
+    lessThan(value, inputRange[offset + 1]),
+    interpolateInternalSingle(value, inputRange, outputRange, offset),
+    interpolateInternal(value, inputRange, outputRange, offset + 1)
+  );
+}
+
+export const Extrapolate = {
+  EXTEND: 'EXTEND',
```

Comment:
Can you verify this work is based on the most recent version? I changed these values to be lowercased to match Animated (also you use lowercase in you example app.

Also it appears that this commit removes `color` nodes that I landed in derived.js

Reply:
@kmagiera 

- Changed the values to lowercase.
- Changed the examples to use the constants instead of using strings.
- Added `color.js`.

Tested the examples, everything seems to be working.